src/patches/resource-extractor/resourceExtractor.ts

```diff
@@ -440,30 +440,39 @@ export class ResourceExtractor implements IAspect {
    */
   private exportValue(stack: Stack, name: string, value: any): string {
     const stackName = stack.stackName;
-    const shareName = `${stackName}:${name}`;
+    let shareName = `${stackName}:${name}`;
+
+    // Intrinsic Function will resolve the value upon deployment
+    const intrinsic = stack.resolve(value);
+
+    // Support for multiple references to the same object
+    if ('Fn::GetAtt' in intrinsic) {
+      const attribute = intrinsic['Fn::GetAtt'][1];
+      shareName = shareName + `:${attribute}`;
+    }
 
     if (this.valueShareMethod == ResourceExtractorShareMethod.SSM_PARAMETER) {
-      const paramName = `/${shareName.replace(':', '/')}`;
-      const paramLogicalId = `p${name}`;
+      shareName = shareName.replace(':', '/').replace(':', '/');
```

Comment:
do you need both `replace`s here?

Reply:
For some reason, yes - it doesn't work with just one. I was confused myself as to the why..

Reply:
can you try

```ts
shareName = shareName.replaceAll(':', '/')
```

Reply:
Thanks, fixed